typescript/common.ts

```diff
@@ -1,5 +1,5 @@
 // From https://github.com/microsoft/TypeScript/issues/1897
-export type JSONPrimitive = string | number | boolean | null | unknown;
+export type JSONPrimitive = string | number | boolean | null | undefined;
 export type JSONValue = JSONPrimitive | JSONObject | JSONArray;
-export type JSONObject = { [member: string]: JSONValue };
+export type JSONObject = { [member: string]: JSONValue | any };
```

Comment:
I left a comment in https://github.com/lastmile-ai/aiconfig/pull/631 (note, your schema typing PR was still in draft but merged) but think we should reconsider this. Ideally JSONPrimitive should not support unknown and JSONObject shouldn't support any

Reply:
The issue is that there are many assignments to properties where type checking complains unless JSONObject is Record<string, any>.

If you remove the any union from JSONObject and open openai.ts, it'll show the issues. It's still a step in the right direction from  having unknown in JSONValue